clang/lib/Frontend/PrintPreprocessedOutput.cpp

```diff
@@ -878,6 +960,27 @@ static void PrintPreprocessedTokens(Preprocessor &PP, Token &Tok,
       std::string Name = M->getFullModuleName();
       Callbacks->OS->write(Name.data(), Name.size());
       Callbacks->HandleNewlinesInToken(Name.data(), Name.size());
+    } else if (Tok.is(tok::annot_embed)) {
```

Comment:
This doesn't emit whitespace before the list of integers. I believe the cause is that HandleWhitespaceBeforeTok needs to have `annot_embed` added to the list of special cases on line 763, above.

Test:
```
printf '\x01' > hello
printf 'foo\n#embed "hello"\n' > hello.c
build/bin/clang -E -o - -std=c23 hello.c
```

Expected: `foo 1` Got: `foo1`

Reply:
Ok, done, though I had to modify `TokenConcatenation::AvoidConcat` too.